sw/device/tests/crypto/aes_gcm_testutils.c

```diff
@@ -0,0 +1,104 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/tests/crypto/aes_gcm_testutils.h"
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/crypto/drivers/aes.h"
+#include "sw/device/lib/crypto/impl/aes_gcm/aes_gcm.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+
+uint64_t t_start;
+uint64_t t_end;
+/**
+ * Start a cycle-count timing profile.
+ */
+static void profile_start() { t_start = ibex_mcycle_read(); }
+
+/**
+ * End a cycle-count timing profile.
+ *
+ * Call `profile_start()` first.
+ */
+static void profile_end(const char *profile_name) {
+  t_end = ibex_mcycle_read();
+  uint32_t cycles = t_end - t_start;
+  LOG_INFO("%s took %d cycles", profile_name, cycles);
+}
+
+uint32_t call_aes_gcm_encrypt(aes_gcm_test_t test) {
+  uint8_t actual_ciphertext[test.plaintext_len];
+  uint8_t actual_tag[kAesGcmTagNumBytes];
+
+  // Construct AES key. Construct key shares by setting first share to full key
+  // and second share to 0. (Note: this is not a secure construction! But it
+  // makes debugging tests easier because there is only one thing to print.)
+  const uint32_t share1[8] = {0};
+  const aes_key_t test_key = {
+      .mode = kAesCipherModeCtr,
+      .sideload = kHardenedBoolFalse,
+      .key_len = test.key_len,
+      .key_shares = {test.key, share1},
+  };
+
+  // Call encrypt() with a cycle count timing profile.
+  profile_start();
+  aes_error_t err = aes_gcm_encrypt(
+      test_key, test.iv_len, test.iv, test.plaintext_len, test.plaintext,
+      test.aad_len, test.aad, actual_ciphertext, actual_tag);
+  profile_end("aes_gcm_encrypt");
+  uint32_t cycles = t_end - t_start;
```

Comment:
Any reason for `t_start` and `t_end` to be global? I feel like it would be clearer like this:

```c
  uint64_t start_time = profile_start();
  // ...
  uint64_t duration = profile_end(start_time);
```

Reply:
You're right, that's better! I've changed the code to match this structure instead.